src/processor/reconnection-graph.service.ts

```diff
@@ -430,12 +419,75 @@ export class ReconnectionGraphService {
     return dsnpKeys;
   }
 
-  async processChainEvents(promises: Promise<ParsedEventResult>[], dsnpUserId: MessageSourceId): Promise<void> {
-    // loop over promises and wait for all to resolve
-    for (const promise of promises) {
-      const [event, eventMap] = await promise;
+  async sendAndProcessChainEvents(
+    dsnpUserId: MessageSourceId,
+    providerId: ProviderId,
+    providerKeys: KeyringPair,
+    graphKeyPairs: ProviderKeyPair[],
+    batchesMap: SubmittableExtrinsic<'rxjs', ISubmittableResult>[][],
+    ): Promise<void> {
+    try {
+      // iterate over batches and send them to the chain
+      batchesMap.forEach(async (batch, batchIndex) => {
+        await this.processSingleBatch(dsnpUserId, providerId, providerKeys, graphKeyPairs, batch);
+      });
+    } catch (e) {
+      this.logger.error(`Error processing batches for ${dsnpUserId.toString()}: ${e}`);
+      throw e;
+    }
+  }
+
+  async processSingleBatch(
+    dsnpUserId: MessageSourceId,
+    provideId: ProviderId,
+    providerKeys: KeyringPair,
+    graphKeyPairs: ProviderKeyPair[],
+    batch: SubmittableExtrinsic<'rxjs', ISubmittableResult>[]
+    ): Promise<void> {
+    try {
+      const [event, eventMap] = await this.blockchainService.createExtrinsic(
+        { pallet: 'frequencyTxPayment', extrinsic: 'payWithCapacityBatchAll' }, 
+        { eventPallet: 'utility', event: 'BatchCompleted' },
+        providerKeys, batch).signAndSend();
+      
       if (!event) {
-        throw new Error(`Error submitting extrinsic`);
+        // if we dont get any events, covering any unexpected connection errors
+        throw new Error(`No events were found for ${dsnpUserId.toString()}`);
+      }
+
+      if(!this.blockchainService.api.events.utility.BatchCompleted.is(event)) {
+        this.logger.warn(`Batch failed event found for ${dsnpUserId.toString()}: ${event}`);
+        if(this.blockchainService.api.events.utility.BatchInterrupted.is(event)) {
+          // Since frequency extrinsics does pre-flight checks via signed extension, we should not see any other events
+          // Any error from such pre-flight checks are caught in the catch block
+          // BatchInterrupted is caused during tx execution and any error arising from
+          // stateful storage would mean our configuration is wrong
+          this.logger.warn(`Unexpected event found for ${dsnpUserId.toString()}`);
+          this.logger.warn(event);
+          this.logger.warn(eventMap);
+          throw new Error(`Batch interrupted event found for ${dsnpUserId.toString()}`);
+        }
+      }
```

Comment:
Since `BatchInteruppted` would essentially mean that config/env is wrong like schemaid etc.